enzyme/Enzyme/GradientUtils.cpp

```diff
@@ -3821,6 +3821,10 @@ Value *GradientUtils::invertPointerM(Value *const oval, IRBuilder<> &BuilderM,
       }
     }
 
+    if (auto arg = dyn_cast<ConstantExpr>(oval)) {
+      return oval;
```

Comment:
You might also want to handle the vector mode case here. E.g. wrap this inside of applyChainRule.